File: frontend/lib/api/dividends.ts
import { apiFetch } from './client';

export interface DividendEntry {
  date: string;
  amount_chf: number;
}

export interface DividendData {
  ticker: string;
  last_dividend_chf: number | null;
  ex_date: string | null;
  dividend_yield_pct: number | null;
  history: DividendEntry[];
  disclaimer: string;
}

export function getDividends(ticker: string): Promise<DividendData> {
  return apiFetch<DividendData>(`/api/v1/stocks/${ticker}/dividends`);
}

File: backend/interfaces/rest/schemas/dividends.py
"""Pydantic-Schema für Dividenden-Response."""

from __future__ import annotations

from pydantic import BaseModel


class DividendEntry(BaseModel):
    """Einzelne Dividendenausschüttung."""

    date: str
    amount_chf: float


class DividendResponse(BaseModel):
    """Response-Schema für GET /api/v1/stocks/{ticker}/dividends."""

    ticker: str
    last_dividend_chf: float | None
    ex_date: str | None
    dividend_yield_pct: float | None
    history: list[DividendEntry]
    disclaimer: str
